pvlib/horizon.py

```diff
@@ -0,0 +1,685 @@
+
+    direction : numeric
+        The direction along which the elevation angle is to be calculated in
+        decimal degrees. The convention is defined as degrees
+        east of north (e.g. North = 0, South=180 East = 90, West = 270).
+
+    Returns
+    --------
+
+    elevation_angle : numeric
+        The angle between the surface of the tilted plane and the horizontal
+        when looking in the specified direction. Given in degrees above the
+        horizontal and limited to be non-negative.
+
+
+    [1] doi.org/10.1016/j.solener.2014.09.037
+
+    """
+    tilt = np.radians(surface_tilt)
+    bearing = np.radians(direction - surface_azimuth - 180.0)
+
+    declination = np.degrees(np.arctan(1.0/np.tan(tilt)/np.cos(bearing)))
+    mask = (declination <= 0)
+    elevation_angle = 90.0 - declination
+    elevation_angle[mask] = 0.0
+
+    return elevation_angle
+
+
+def calculate_dtf(horizon_azimuths, horizon_angles,
+                  surface_tilt, surface_azimuth):
+    """
+    Calculate the diffuse tilt factor for a tilted plane that is adjusted
+    with for horizon profile. The idea for a diffuse tilt factor is explained
+    in [1].
+
+    Parameters
+    ----------
+    horizon_azimuths: numeric
+        Azimuth values for points that define the horizon profile. The ith
+        element in this array corresponds to the ith element in horizon_angles.
+
+    horizon_angles: numeric
+        Elevation angle values for points that define the horizon profile. The
+        elevation angle of the horizon is the angle that the horizon makes with
+        the horizontal. It is given in degrees above the horizontal. The ith
+        element in this array corresponds to the ith element in
+        horizon_azimuths.
+
+    surface_tilt : numeric
+        Surface tilt angles in decimal degrees. surface_tilt must be >=0
+        and <=180. The tilt angle is defined as degrees from horizontal
+        (e.g. surface facing up = 0, surface facing horizon = 90)
+
+    surface_azimuth : numeric
+        Surface azimuth angles in decimal degrees. surface_azimuth must
+        be >=0 and <=360. The azimuth convention is defined as degrees
+        east of north (e.g. North = 0, South=180 East = 90, West = 270).
+
+    Returns
+    -------
+    dtf: numeric
+        The diffuse tilt factor that can be multiplied with the diffuse
+        horizontal irradiance (DHI) to get the incident irradiance from
+        the sky that is adjusted for the horizon profile and the tilt of
+        the plane.
+
+    Notes
+    _____
+
+    The dtf in this method is calculated by approximating the surface integral
+    over the visible section of the sky dome. The integrand of the surface
+    integral is the cosine of the angle between the incoming radiation and the
+    vector normal to the surface. The method calculates a sum of integrations
+    from the "peak" of the sky dome down to the elevation angle of the horizon.
```

Comment:
Could you mention the relevant section or equation(s) in [1]?

Reply:
Do you mean [2]?

Reply:
I only see one reference...

Reply:
my bad. latest changes have the new reference. Old one was wrong.

Reply:
Ok, I read this paper too.  Which equation(s) are you implementing?